llvm/include/llvm/IR/Function.h

```diff
@@ -75,6 +75,38 @@ class LLVM_EXTERNAL_VISIBILITY Function : public GlobalObject,
 private:
   // Important things that make up a function!
   BasicBlockListType BasicBlocks;         ///< The basic blocks
+
+  // Basic blocks need to get their number when added to a function.
+  friend void BasicBlock::setParent(Function *);
+  unsigned NextBlockNum = 0;
+  /// Epoch of block numbers. (Could be shrinked to uint8_t if required.)
+  unsigned BlockNumEpoch = 0;
+
+public:
```

Comment:
Can you please move these functions after all the variable declarations? It makes it hard to see what the layout of the structure is if everything is interleaved.